aten/src/ATen/core/Vitals.h

```diff
@@ -40,11 +44,36 @@ struct TorchVital {
   }
 };
 
+// A way to access vitals by string names instead of by global reference.
+// This enables access to vitals from the PythonAPI.
+class TORCH_API APIVitals
+{
+public:
+    // Set any vital sign that was added to the map.
+    bool setVital(const std::string &vital_name, const std::string &attr_name, const std::string &value);
+
+    APIVitals(): name_map_() { }
+
+    // Ensure this stays a singleton
+    APIVitals(APIVitals const& other) = delete;
+    APIVitals(APIVitals&& other) = delete;
+    APIVitals& operator=(const APIVitals&) = delete;
+    APIVitals& operator=(APIVitals&&) = delete;
+
+private:
+  // weak_ptr keeps semantics where vitals are print-on-delete.
+    std::map<std::string, TorchVital> name_map_;
```

Comment:
why is this an std::map and not std::unordered_map or, better yet, flat_hash_map? Also, the comment above is unclear, can you please expand it?

Reply:
Moved to flat_hash_map! Removed comment, no longer applies to current code :)